From c8b2932009bed1066e076473f4feaf3b6c878bed Mon Sep 17 00:00:00 2001
From: Manjunath Kumatagi <mkumatag@in.ibm.com>
Date: Mon, 31 Aug 2020 23:20:49 +0530
Subject: Skip TestLoopbackHostPortIPv6 run on non-IPv6 env

Cherry-pick of https://github.com/kubernetes/kubernetes/pull/94376,
which is included in upstream Kubernetes 1.20.

Fixes TestLoopbackHostPortIPv6 test failure if there is no IPv6
loopback device configured.

Signed-off-by: Kirsten Schumy <ksschumy@amazon.com>
---
 .../src/k8s.io/apiserver/pkg/server/config_selfclient_test.go   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go b/staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go
index 19e975ffb6a..4d811c64104 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go
@@ -60,7 +60,7 @@ func TestLoopbackHostPortIPv6(t *testing.T) {
 		t.Fatalf("fail to enumerate network interface, %s", err)
 	}
 	if !ipv6 {
-		t.Fatalf("no ipv6 loopback interface")
+		t.Skip("no ipv6 loopback interface")
 	}
 
 	host, port, err := LoopbackHostPort("[ff06:0:0:0:0:0:0:c3]:443")
-- 
2.35.1

